### PR TITLE
Tweaks for check-typo

### PR DESCRIPTION
**Pull request by dra27**
This is a whole load of bits and bobs which, combined with #1288, #1827, #1828 and #1830 ~a small tweak for `Makefile` which I'm just checking works~, means that the entire repository passes the `tools/check-typo` script.

Various points:
 - [x] @Octachron - missing copyright header on `ocamldoc/stdlib_non_prefixed/Makefile` OK?
 - [x] @xavierleroy & @damiendoligez - various copyright headers restored as some support files had been moved out of the `tests` directory. Alternative would be to add the support directories to the `missing-header` rule.
 - [x] @shindere - I've tweaked several Inria CI scripts for `long-line`. Will they be exercised simply by doing a precheck run on this branch?
 - [x] @gasche - consider your wrists slapped for submitting #1515 with no copyright headers on your files :wink: Are they correct?

Unlike #1288, this branch does tweak code in order to satisfy `tools/check-typo` so it does all need reviewing (just in case that wasn't immediately apparent).

### CODE REVIEW

**Comment by gasche** [comment]
Yes, the copyright headers are fine. Thanks, and sorry for the omission.

**Comment by Octachron** [comment]
The ocamldoc copyright headers look fine.

**Comment by dra27** [comment]
@shindere - with reference to the suggestion in #1632, I've noted in `.gitattributes` that `run` and `check-program-output` are ocamltest actions which have scripts. Given the large number of actions which are available, the fact that only two presently have scripts used for them and of course the fact that more actions may be added, I propose that we only list the actions which we use for scripts in `.gitattributes`?

**Comment by shindere** [comment]
David Allsopp (2018/06/08 09:00 -0700):
>  - [ ] @shindere - I've tweaked several Inria CI scripts for
>  `long-line`. Will they be exercised simply by doing a precheck run on
>  this branch?

I don't think so. At the moment, Inria's CI does not run check-typo.

I think we could add a job for that once all your related PRs are
merged, I would be happy to set-up cuch a job.

Re: the renaming of scripts, well to be honest I am not 100% found of
the precheck extension, whihc I find a bit meaningless, although I
understand it helps ot have a dedicated extension.

One thing whihc is not completely solved is that, when calling scripts
on Windows, it's a bit unclear how the file paths should be encoded: the
windows or the Unix way. Perhaps the extensions of scripts could be used
for that and we could have one extension to tell ocamltest to export
filepaths the native (Windows) way and another extension to request
paths being exported the Unix way?

**Comment by shindere** [comment]
David Allsopp (2018/06/10 07:52 -0700):
> @shindere - with reference to the suggestion in #1632, I've noted in
> `.gitattributes` that `run` and `check-program-output` are ocamltest
> actions which have scripts. Given the large number of actions which
> are available, the fact that only two presently have scripts used for
> them and of course the fact that more actions may be added, I propose
> that we only list the actions which we use for scripts in
> `.gitattributes`?

Sure, that's fine with me. These scripts are actually hooks in ocamltest
terminology. :)

File: tools/ci/inria/extra-checks [hunk 1]
```diff
@@ -21,7 +21,8 @@
 # In this context, it is necessary to skip a few tests whose behaviour
 # is modified by the instrumentation:
 
-export OCAMLTEST_SKIP_TESTS="tests/afl-instrumentation/afltest.ml tests/runtime-errors/stackoverflow.ml"
+export OCAMLTEST_SKIP_TESTS="tests/afl-instrumentation/afltest.ml "\
```
Comment (damiendoligez) [inline tools/ci/inria/extra-checks]: This looks a bit strange. Why not just split the line with a backslash inside the quotes?

**Review by damiendoligez — APPROVED** [review]
All these changes look good to me.

File: tools/ci/inria/extra-checks (hunk 1 above)
Comment (dra27) [reply tools/ci/inria/extra-checks]: I think I got in a muddle - that’s the version where you can (sometimes) then indent the next line too. I’ll change it when I rebase.

**Comment by dra27** [comment]
@shindere - I meant when are the scripts I’ve changed here run (extra-checks and bootstrap)... in case my continuation lines have broken something!

**Comment by dra27** [comment]
@shindere - could just use .sh instead of .precheck?

**Comment by shindere** [comment]
David Allsopp (2018/06/12 00:27 -0700):
> @shindere - I meant when are the scripts I’ve changed here run
> (extra-checks and bootstrap)... in case my continuation lines have
> broken something!

Oh okay, sorry. At themoment, bootstrap is run only on trunk because the
PR that fixes theprocedure has not been merged into 4.07.

The extra-checks script is run on both trunk and 4.07 branches.

**Comment by shindere** [comment]
David Allsopp (2018/06/12 00:28 -0700):
> @shindere - could just use .sh instead of .precheck?

Sure. I find it a bit unclean to have extensions, but why not.

**Comment by dra27** [comment]
> > @shindere - could just use .sh instead of .precheck?
>
> Sure. I find it a bit unclean to have extensions, but why not.

Heh, heh - from my OS, I find the lack of them unclean :wink: More seriously, it's useful discipline for the repo just as the `eol=lf` gitattribute is applied by wildcard.

**Comment by dra27** [comment]
> Oh okay, sorry. At themoment, bootstrap is run only on trunk because the
> PR that fixes theprocedure has not been merged into 4.07.
> 
> The extra-checks script is run on both trunk and 4.07 branches.

OK, thanks - that means we'll know quickly if something has gone wrong (I'll get the ci-notifications too!)

**Comment by shindere** [comment]
David Allsopp (2018/06/12 02:11 -0700):
> Heh, heh - from my OS, I find the lack of them unclean :wink:

That's a good point, you're right! All this is very relative.

> More seriously, it's useful discipline for the repo just as the
> `eol=lf` gitattribute is applied by wildcard.

Yeah, I agree. Are you willing to rename the files or shall I do it?

More fundamentally, can you think about a nice way of distingishing the
scripts for which the file paths should be exported natively / the Unix
way? Should that be based on the extension of the script, or shall we
have an ocamltest variable to specify this, or is there yet another way?

**Comment by shindere** [comment]
David Allsopp (2018/06/12 02:12 -0700):
> > Oh okay, sorry. At themoment, bootstrap is run only on trunk because the
> > PR that fixes theprocedure has not been merged into 4.07.
> > 
> > The extra-checks script is run on both trunk and 4.07 branches.
> 
> OK, thanks - that means we'll know quickly if something has gone wrong
> (I'll get the ci-notifications too!)

Very quickly,yes.

**Comment by dra27** [comment]
All changes addressed, I hope - @shindere, I need to look at the paths question in more detail, as I don't (yet) fully understand the problem. I expect that it depends on what is responsible for executing the script, but let me have a closer look later.

**Comment by shindere** [comment]
Sure, @dra27! We can also discuss directly if you like.

**Comment by dra27** [comment]
Everyone OK for this to be merged?

**Comment by shindere** [comment]
David Allsopp (2018/06/12 09:29 -0700):
> Everyone OK for this to be merged?

+1 !

**Pull request merged**